lib/matplotlib/backend_bases.py

```diff
@@ -2791,6 +2791,12 @@ def mouse_move(self, event):
             except (ValueError, OverflowError):
                 pass
             else:
+                ax = event.inaxes
+                artists = ax.artists + ax.images + ax.lines
```

Comment:
I think to first order, just ignore the compositing and use the top image. 

The in-place compositing is only for some of the vector backends which can't do it them selves (ex pdf) to try and save space (so you don't have multiple overlapping images) but it leads to other bugs as all images get smashed to the same zorder relative to the other lines etc.   In Agg (which backs all but 3 (gtkcairo, macsox and windows) of the interactive backends) the whole `if _do_composite` block is skipped and the compsiting is taken care of at the c++ layer.